src/code/playback.c

```diff
@@ -0,0 +1,786 @@
+        return aNote;
+    }
+    rNote->playbackState.wantedParentLayer = seqLayer;
+    rNote->playbackState.priority = seqLayer->seqChannel->notePriority;
+    return rNote;
+}
+
+Note* alloc_note(SequenceChannelLayer* seqLayer) {
+    Note* ret;
+    u32 policy = seqLayer->seqChannel->noteAllocPolicy;
+
+    if (policy & 1) {
+        ret = seqLayer->note;
+        if (ret != NULL && ret->playbackState.prevParentLayer == seqLayer &&
+            ret->playbackState.wantedParentLayer == NO_LAYER) {
+            note_release_and_take_ownership(ret, seqLayer);
+            audio_list_remove(&ret->listItem);
+            audio_list_push_back(&ret->listItem.pool->releasing, &ret->listItem);
+            return ret;
+        }
+    }
+
+    if (policy & 2) {
+        if (!(ret = alloc_note_from_disabled(&seqLayer->seqChannel->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_active(&seqLayer->seqChannel->notePool, seqLayer))) {
+            goto null_return;
+        }
+        return ret;
+    }
+
+    if (policy & 4) {
+        if (!(ret = alloc_note_from_disabled(&seqLayer->seqChannel->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_disabled(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_active(&seqLayer->seqChannel->notePool, seqLayer)) &&
+            !(ret = alloc_note_from_active(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer))) {
+            goto null_return;
+        }
+        return ret;
+    }
+
+    if (policy & 8) {
+        if (!(ret = alloc_note_from_disabled(&gNoteFreeLists, seqLayer)) &&
+            !(ret = alloc_note_from_decaying(&gNoteFreeLists, seqLayer)) &&
+            !(ret = alloc_note_from_active(&gNoteFreeLists, seqLayer))) {
+            goto null_return;
+        }
+        return ret;
+    }
+
+    if (!(ret = alloc_note_from_disabled(&seqLayer->seqChannel->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_disabled(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_disabled(&gNoteFreeLists, seqLayer)) &&
+        !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_decaying(&gNoteFreeLists, seqLayer)) &&
+        !(ret = alloc_note_from_active(&seqLayer->seqChannel->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_active(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_active(&gNoteFreeLists, seqLayer))) {
+        goto null_return;
+    }
+    return ret;
+
+null_return:
```

Comment:
Replacing the goto null_returns with
```
seqLayer->unusedEu0b8 = 1;
return NULL;
```

does not match?

Reply:
So I'll give this one a retry, but that is actually what SM64 has but it didn't match in this situation. For some reason it doesn't know to merge the tails here (perhaps there's a compiler flag for this?)

Reply:
Yeah, just checked and it doesn't merge tails for optimization. Maybe there's a way to that would trigger it naturally, but I have been unable to.